### PR TITLE
Add custom domain support to Nest CLI

### PR DESCRIPTION
**Pull request by dispherical**

### CODE REVIEW

File: cli/add_domain.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+REAL_CNAME=$(dig +short -t CNAME "$DOMAIN")
+
+if [ "$NEEDED_CNAME" != "$REAL_CNAME" ]; then
+    echo "The domain $DOMAIN does not have the correct CNAME. Expected: $NEEDED_CNAME, Actual: $REAL_CNAME"   
+	exit 1 
+fi
+
+# Check for existance of subdomain
+if grep $DOMAIN /etc/caddy/Caddyfile &> /dev/null; then
+	echo "You already have this subdomain ($DOMAIN)!"
```
Comment (polypixeldev) [inline cli/add_domain.sh]:
```suggestion
	echo "You already have this custom domain ($DOMAIN)!"
```

**Comment by dispherical** [comment]
Added FQDN validation + updated error message so it shows "None" instead of a blank CNAME record

File: cli/add_domain.sh [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    echo "Invalid FQDN"
+    exit 1
+fi
+
+if [ "$NEST_USER" = "root" ]; then
+	echo "Command cannot be run as root!"
+	exit 1
+fi
+
+NEEDED_CNAME="$USER.hackclub.app"
```
Comment (polypixeldev) [inline cli/add_domain.sh]:
```suggestion
NEEDED_CNAME="$USER.hackclub.app."
```
From my testing, the `dig` command will return the full DNS label with a trailing `.`

**Review by polypixeldev — CHANGES_REQUESTED** [review]
Couple quick things, otherwise LGTM!

**Review by polypixeldev — APPROVED** [review]
LGTM!

**Pull request merged**